### DISCUSSION TITLE
Alembic Detect

### DISCUSSION THREAD

**Discussion by hamedyosefian** [sqlalchemy/alembic#1134] [Usage Questions]
How detect everything with alembic auto generate ??
Similar change enum or change type field and ...
i have many model and can't check everything in each migration file

**Comment by zzzeek** [comment]
enum detection isn't implemented right now, however for general type comparison you can enable [compare_type](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#comparing-types)

**Reply by hamedyosefian** [reply]
Changes of table name and Changes of column name no way to detect ?

**Reply by pykenny** [reply]
It's difficult to define "renaming" between two schemas/columns. [Git](https://stackoverflow.com/questions/24163336/why-does-git-treat-this-delete-new-as-a-rename) tries to make some inference about "file renaming" (where the threshold can be tuned) that happened within a commit, but that only affects terminal output, and doesn't change the fact that the 'renaming' operation shown in output is actually combination of deleting one file, plus addition of another.

**Reply by pykenny** [reply]
If model/table `A` disappeared, while model/table `B` having the same schema emerged at the meantime, there are two possibilities:
 - `A` got renamed as `B`, and `B` inherits all the records from `A`.
 - `A` got removed entirely, and `B` was set up to store some other stuff.

And it's impossible to tell which one's actually happening by merely comparing model definitions. Alembic makes guess of the second situation by default.

**Reply by hamedyosefian** [reply]
my big problem when change enum value not detect how to solve it ?

**Reply by CaselIT** [reply]
that's currently not supported. there are already some issue regarding it https://github.com/sqlalchemy/alembic/issues?q=is%3Aopen+is%3Aissue+label%3A%22autogenerate+for+enums%22
